src/main/java/org/wso2/carbon/inbound/salesforce/poll/LoginHelper.java

```diff
@@ -0,0 +1,185 @@
+                    reading = true;
+                    break;
+                case "serverUrl":
+                    reading = true;
+                    break;
+                case "faultstring":
+                    reading = true;
+                    break;
+                default:
+            }
+        }
+    }
+
+    public static final String COMETD_REPLAY = "/cometd/";
+    public static final String COMETD_REPLAY_OLD = "/cometd/replay/";
+    static final String LOGIN_ENDPOINT = "https://login.salesforce.com";
+    private static final String ENV_END = "</soapenv:Body></soapenv:Envelope>";
+    private static final String ENV_START =
+            "<soapenv:Envelope xmlns:soapenv='http://schemas.xmlsoap.org/soap/envelope/' "
+                    + "xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' "
+                    + "xmlns:urn='urn:partner.soap.sforce.com'><soapenv:Body>";
+    // The enterprise SOAP API endpoint used for the login call
+    private static final String SERVICES_SOAP_PARTNER_ENDPOINT =
+            "/services/Soap/u/" + SalesforceDataHolderObject.getSoapApiVersion() + "/";
+
+    public static BayeuxParameters login(String username, String password) throws Exception {
+
+        return login(new URL(LOGIN_ENDPOINT), username, password);
+    }
+
+    public static BayeuxParameters login(String username, String password, BayeuxParameters params) throws Exception {
+
+        return login(new URL(LOGIN_ENDPOINT), username, password, params);
+    }
+
+    public static BayeuxParameters login(URL loginEndpoint, String username, String password) throws Exception {
+
+        return login(loginEndpoint, username, password, new BayeuxParameters() {
+            @Override
+            public String bearerToken() {
+
+                throw new IllegalStateException("Have not authenticated");
+            }
+
+            @Override
+            public URL endpoint() {
+
+                throw new IllegalStateException("Have not established replay endpoint");
+            }
+        });
+    }
+
+    public static BayeuxParameters login(URL loginEndpoint, String username, String password,
+                                         BayeuxParameters parameters) throws Exception {
+
+        HttpClient client = new HttpClient(parameters.sslContextFactory());
+        try {
+            client.getProxyConfiguration().getProxies().addAll(parameters.proxies());
+            client.start();
+            URL endpoint = new URL(loginEndpoint, getSoapUri());
+            Request post = client.POST(endpoint.toURI());
+            post.content(
+                    new ByteBufferContentProvider("text/xml", ByteBuffer.wrap(soapXmlForLogin(username,
+                            password))));
+            post.header("SOAPAction", "''");
+            post.header("PrettyPrint", "Yes");
+            ContentResponse response = post.send();
+            SAXParserFactory spf = SAXParserFactory.newInstance();

```

Comment:
Better to leave empty line to improve the readability.

Reply:
line added